feat(socialaccount) Add Sharefile provider

Cabalist opened the pull request:
This adds a ShareFIle (https://www.sharefile.com/) Oauth2 provider.   We are currently using this in our app.  Was a pretty painless addition but let me know if there is something more needed to get this merged. 

Thanks for such a great library! 


## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
 - [x] All Python code must be 100% pep8 and isort clean.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [ ] If your changes are significant, please update `ChangeLog.rst`.
 - [x] Feel free to add yourself to `AUTHORS`.
 
 ## Provider Specifics
 
 In case you add a new provider:
 
- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

Cabalist commented on docs/providers.rst:
```diff
@@ -1142,7 +1143,7 @@ SCOPE:
 
 QuickBooks
-------
+----------
```
My editor was complaining about a bunch of warnings and errors in this file so I tidied them up

Cabalist commented:
FYI Looks like the errors for this are related to Django 2.2 and sqlite and not the PR code.

pennersr commented:
The Django 2.2 related issues have been fixed on master. But this PR does not pass the pep8 checker, can you have a look?

Cabalist commented:
Thanks!   I've rebased and fixed some trailing white space.  It looks like I am hun on the 79 character line length in flake8. 

Double checking that is requirement.  :)

Cabalist commented:
I've got flake8 passing but by doing so reduced Coverage by 0.09%. .   And an unrelated Cern test seems to be failing on one instance but not the rest.   CI whack-a-mole....  :)


Sorry for all the force pushes.  I really need to learn how to squash commits.
